libs/payments/ui/src/lib/client/components/CancelSubscription/index.tsx

```diff
@@ -0,0 +1,270 @@
+              }}
+            >
+              <p className="leading-6 py-3" id="cancel-subscription-desc">
+                You will no longer be able to use {productName} after{' '}
+                {currentPeriodEndLongFallback}, the last day of your billing
+                cycle.
+              </p>
+            </Localized>
+            <Form.Label asChild className="cursor-pointer my-3">
+              <label htmlFor="cancelAccess" className="flex items-start gap-4">
+                <Form.Control asChild>
+                  <input
+                    id="cancelAccess"
+                    name="cancelAccess"
+                    type="checkbox"
+                    className="ml-1 mt-1 grow-0 shrink-0 basis-4 scale-150 cursor-pointer"
+                    onChange={(e) => setCheckedState(e.target.checked)}
+                    required
+                  />
+                </Form.Control>
+                <Localized
+                  id="subscription-content-cancel-access-message"
+                  vars={{
+                    productName,
+                    currentPeriodEnd: currentPeriodEndLongFallback,
+                  }}
+                >
+                  <span className="leading-6">
+                    Cancel my access and my saved information within{' '}
+                    {productName} on {currentPeriodEndLongFallback}
+                  </span>
+                </Localized>
+              </label>
+            </Form.Label>
+            {showCancelActionError && (
+              <Form.Message asChild>
+                <Localized id="subscription-content-cancel-action-error">
+                  <p className="mt-1 text-alert-red font-normal" role="alert">
+                    An unexpected error occurred. Please try again.
+                  </p>
+                </Localized>
+              </Form.Message>
+            )}
+          </Form.Field>
+          <div className="w-full flex flex-col gap-3 items-center pt-3">
+            <Form.Submit asChild>
+              <SubmitButton
+                className="font-bold h-14 w-full"
+                variant={ButtonVariant.SubscriptionManagementError}
+                disabled={!checkedState}
+                aria-label={`Cancel your subscription to ${productName}`}
+              >
+                <Localized
+                  id="cancel-subscription-button-cancel-subscription"
+                  vars={{ productName }}
+                  attrs={{ 'aria-label': true }}
+                >
+                  <span>Cancel subscription</span>
+                </Localized>
+              </SubmitButton>
+            </Form.Submit>
+            <Link
+              className="border box-border font-bold font-header h-14 items-center justify-center rounded text-center py-2 px-5 bg-grey-10 border-grey-200 hover:bg-grey-50 flex w-full"
+              onClick={(e) => {
+                e.preventDefault();
+                setLoading(true);
```

Comment:
Is there a reason this and the other buttons set isLoading to true when they just push to the router anyway?

Reply:
Per discussion, in case of delay when navigating back to Subscription Management, a loading animation will appear.